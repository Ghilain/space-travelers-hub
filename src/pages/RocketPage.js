import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from '../components/UI/rockets/Rocket';

const RocketPage = () => {
  const rockets = useSelector((state) => state.rockets);
  const rocketElements = rockets.map((rocket) => (
    <Rocket
      key={rocket.id}
      id={rocket.id}
      img={rocket.img}
      description={rocket.description}
      title={rocket.rocket_name}
      active={rocket.active}
    />
  ));
  return (
    <ul>
      { rocketElements }
    </ul>
  );
};
export default RocketPage;
