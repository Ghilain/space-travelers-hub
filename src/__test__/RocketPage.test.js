import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';
import RocketPage from '../pages/RocketPage';

const rockets = [
  {
    description: 'Description example',
    id: 1,
    rocket_name: 'First mission',
    active: true,
  },
];

describe('testing the rendering of missions', () => {
  it('', () => {
    const { container } = render(
      <Provider store={store}>
        <RocketPage key={rockets[0]} rockets={rockets[0]} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
