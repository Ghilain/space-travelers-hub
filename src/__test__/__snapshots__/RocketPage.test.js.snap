// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of missions  1`] = `
<div>
  <ul />
</div>
`;
